packages/client-config/src/generate_client_config_to_file.ts

```diff
@@ -14,12 +18,28 @@ export const generateClientConfigToFile = async (
   outDir?: string,
   format?: ClientConfigFormat
 ): Promise<void> => {
-  const clientConfigWriter = new ClientConfigWriter();
+  const packageJson = await readPackageJson();
+
+  const clientConfigWriter = new ClientConfigWriter(
+    getClientConfigPath,
+    new ClientConfigFormatter(
+      new ClientConfigConverter(packageJson.name, packageJson.version)
+    )
+  );
 
   const clientConfig = await generateClientConfig(
     credentialProvider,
     backendIdentifier
   );
-  const targetPath = await getClientConfigPath(outDir, format);
-  await clientConfigWriter.writeClientConfig(clientConfig, targetPath);
+  await clientConfigWriter.writeClientConfig(clientConfig, outDir, format);
+};
+
+const readPackageJson = async (): Promise<{
```

Comment:
we have `loadCwdPackageJson` in the cli package. Not sure if this duplication warrants pulling that into `plaform-core` but something to consider

Reply:
I'm doing that as part of my usage metrics PR, will pull it in once this is merged.

Reply:
I looked at `loadCwdPackageJson` but what we need here is really `loadMyPackageJson` not cwd, which is first occurence of this (if we set aside typing of package json).

@Amplifiyer thanks!